src/login/logind-dbus.c

```diff
@@ -2095,13 +2096,17 @@ static int verify_shutdown_creds(
         }
 
         if (blocked) {
-                if (!FLAGS_SET(flags, SD_LOGIND_SKIP_INHIBITORS))
+                PolkitFlags polkit_flags = 0;
+
+                if (!FLAGS_SET(flags, SD_LOGIND_SKIP_INHIBITORS) &&
+                    (uid != 0 || (FLAGS_SET(flags, SD_LOGIND_ROOT_CHECK_INHIBITORS) && offending->mode == INHIBIT_BLOCK_WEAK)))
```

Comment:
the uid check is not right. you are checking in too many cases now. the old check *only* applied to root. But here you say non-root gets refused always.

Reply:
As far as I can see, the behaviour with this is the same as before. This side also only applied to root before. Is there a use case where it would be different, and if so, how?

Reply:
iow, as far as I can tell from testing old/new images, this check here covers one specific case: a block held by non-root, when root tries to reboot. This ensures that with the weak blocker, if the flag is set, the reboot is blocked.

Is there a different workflow that you had in mind?

Reply:
The check is wrong: think about it. in v256 if you are unpriv and a weak block inhibitor is taken by someone else we'd ask PK for authorization to override this. With your patch we don't, we'll just refuse and tell the user to go fuck themselves, without any chance to override it via PK.

Reply:
So in other words uid X has weak block and uid Y tries to reboot, with both X and Y non-zero? Sure let me check

Reply:
mmh on this branch:

first session:

```
test@localhost:~$ systemd-inhibit --what=shutdown --mode=block-weak sleep infinity
```

second session:

```
test2@localhost:~$ systemctl reboot --check-inhibitors no
==== AUTHENTICATING FOR org.freedesktop.login1.reboot-multiple-sessions ====
Authentication is required to reboot the system while other users are logged in.
Authenticating as: root
Password: 
==== AUTHENTICATION COMPLETE ====
```

Reply:
ah nvm that's because systemctl appends SD_LOGIND_SKIP_INHIBITORS

Reply:
Ok updated, now it should match the old behaviour, PTAL